core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionJobPrioritizer.java

```diff
@@ -19,42 +19,120 @@
+
+  static final Pair<Short,Short> SYSTEM_NS_USER =
+      new Pair<>((short) (SYSTEM_NS_SYSTEM_MAX + 1), SYSTEM_NS_USER_MAX);
+  static final Pair<Short,Short> SYSTEM_NS_SYSTEM =
+      new Pair<>((short) (USER_TABLE_USER_MAX + 1), SYSTEM_NS_SYSTEM_MAX);
+
+  static final Pair<Short,Short> USER_TABLE_USER =
+      new Pair<>((short) (USER_TABLE_SYSTEM_MAX + 1), USER_TABLE_USER_MAX);
+  static final Pair<Short,Short> USER_TABLE_SYSTEM =
+      new Pair<>(Short.MIN_VALUE, USER_TABLE_SYSTEM_MAX);
 
-    int prio = totalFiles + compactingFiles;
+  static {
+    // root table
+    SYSTEM_TABLE_RANGES.put(new Pair<>(RootTable.ID, CompactionKind.USER), ROOT_TABLE_USER);
+    SYSTEM_TABLE_RANGES.put(new Pair<>(RootTable.ID, CompactionKind.SYSTEM), ROOT_TABLE_SYSTEM);
 
-    switch (kind) {
-      case USER:
-      case CHOP:
-        // user-initiated compactions will have a positive priority
-        // based on number of files
-        if (prio > Short.MAX_VALUE) {
-          return Short.MAX_VALUE;
+    // metadata table
+    SYSTEM_TABLE_RANGES.put(new Pair<>(MetadataTable.ID, CompactionKind.USER), METADATA_TABLE_USER);
+    SYSTEM_TABLE_RANGES.put(new Pair<>(MetadataTable.ID, CompactionKind.SYSTEM),
+        METADATA_TABLE_SYSTEM);
+
+    // metadata table
+    ACCUMULO_NAMESPACE_RANGES.put(new Pair<>(Namespace.ACCUMULO.id(), CompactionKind.USER),
+        SYSTEM_NS_USER);
+    ACCUMULO_NAMESPACE_RANGES.put(new Pair<>(Namespace.ACCUMULO.id(), CompactionKind.SYSTEM),
+        SYSTEM_NS_SYSTEM);
+  }
+
+  public static short createPriority(NamespaceId nsId, TableId tableId, CompactionKind kind,
+      int totalFiles, int compactingFiles, int maxFilesPerTablet) {
+
+    Objects.requireNonNull(nsId, "nsId cannot be null");
+    Objects.requireNonNull(tableId, "tableId cannot be null");
+    Preconditions.checkArgument(totalFiles >= 0, "totalFiles is negative %s", totalFiles);
+    Preconditions.checkArgument(compactingFiles >= 0, "compactingFiles is negative %s",
+        compactingFiles);
+
+    if (totalFiles > maxFilesPerTablet && kind == CompactionKind.SYSTEM) {
+      int priority =
+          TABLE_OVER_SIZE.getFirst() + compactingFiles + (totalFiles - maxFilesPerTablet);
+      if (priority > Short.MAX_VALUE) {
+        return Short.MAX_VALUE;
+      }
+      return (short) priority;
+    } else {
+      Pair<Short,Short> range = null;
+      if (Namespace.ACCUMULO.id() == nsId) {
+        range = SYSTEM_TABLE_RANGES.get(new Pair<>(tableId, kind));
+        if (range == null) {
+          range = ACCUMULO_NAMESPACE_RANGES.get(new Pair<>(nsId, kind));
         }
-        return (short) prio;
-      case SELECTOR:
-      case SYSTEM:
-        // system-initiated compactions will have a negative priority
-        // starting at -32768 and increasing based on number of files
-        // maxing out at -1
-        if (prio > Short.MAX_VALUE) {
-          return -1;
+      } else {
+        if (kind == CompactionKind.SYSTEM) {
```

Comment:
The following would line up with what the old code was doing.  May be able to revert some changes to one of the unit test with this change.

```suggestion
        if (kind == CompactionKind.SYSTEM || kind == CompactionKind.SELECTOR) {
```